app/src/main/java/com/rockthevote/grommet/ui/registration/review/ReviewAndConfirmFragment.java

```diff
@@ -84,14 +70,54 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     @Override
     public void storeState() {
         ReviewData data = ReviewExtKt.toReviewData(binding);
-        viewModel.storeReviewData(data);
+        viewModel.completeRegistration(data);
     }
 
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         ButterKnife.bind(this, view);
         dialog = new DisclosureAgreementDialogFragment();
+
+        // TODO Observe Registration state
+        viewModel.getRegistrationState().observe(getViewLifecycleOwner(), registrationState -> {
+            if (registrationState instanceof RegistrationState.Complete) {
```

Comment:
I see the need for the completion state now

Reply:
Now I'm fairly certain an abandoned state would be good to implement. We need to track number of abandoned registrations per canvassing session. Incrementing the count when an abandoned state is observed (i.e. when the user cancels the registration) seems logical to me.

Reply:
Where/how can the user end up in an abandoned state? It may be good to have a listener for this state in the Activity so it can be emitted from anywhere within the registration process.

Reply:
they just click "cancel" , it's a dropdown at the top right